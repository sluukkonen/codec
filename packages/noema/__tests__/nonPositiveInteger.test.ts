import { nonPositiveInteger } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should accept non-positive integers", () => {
  expectParseSuccess(nonPositiveInteger, 0)
  expectParseSuccess(nonPositiveInteger, -1)
  expectParseSuccess(nonPositiveInteger, -2)
  expectParseSuccess(nonPositiveInteger, -3)
})

test("should not accept positive integers", () => {
  expectParseFailure(nonPositiveInteger, 1)
  expectParseFailure(nonPositiveInteger, 2)
  expectParseFailure(nonPositiveInteger, 2)
})

test("should not accept non-integers", () => {
  expectParseFailure(nonPositiveInteger, -0.5)
  expectParseFailure(nonPositiveInteger, 0.5)
})

test("should not accept NaN", () => {
  expectParseFailure(nonPositiveInteger, NaN)
})
