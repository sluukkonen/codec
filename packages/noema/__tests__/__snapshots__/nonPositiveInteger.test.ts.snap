// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not accept NaN 1`] = `
[
  {
    "code": "is_nan",
    "path": [],
  },
]
`;

exports[`should not accept non-integers 1`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`should not accept non-integers 2`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`should not accept positive integers 1`] = `
[
  {
    "code": "too_large",
    "max": 0,
    "min": undefined,
    "path": [],
  },
]
`;

exports[`should not accept positive integers 2`] = `
[
  {
    "code": "too_large",
    "max": 0,
    "min": undefined,
    "path": [],
  },
]
`;

exports[`should not accept positive integers 3`] = `
[
  {
    "code": "too_large",
    "max": 0,
    "min": undefined,
    "path": [],
  },
]
`;
